app/Http/Controllers/Setup/Controller.php

```diff
@@ -24,20 +25,26 @@ public function setup() {
           || !is_writable(__DIR__.'/../../../../storage/logs/.gitignore')) {
             return view('setup/permissions');
         }
-
+        if(!empty(env('DATABASE_URL'))) {
+            return redirect(route('setup.database'));
+        }
         return view('setup/start');
     }
 
     public function database() {
         // Check for a Heroku ClearDB config string and set that as the defaults
-        if($url = getenv('CLEARDB_DATABASE_URL')) {
+        $url = env('DATABASE_URL');
+        if(!empty($url)) {
             $db = parse_url($url);
             session([
                 'setup.db_name' => substr($db['path'], 1),
                 'setup.db_host' => $db['host'],
                 'setup.db_username' => $db['user'],
                 'setup.db_password' => $db['pass'],
             ]);
+            if(empty(session('db-error'))) {
+                return redirect(route('setup.test-database-auto'));
```

Comment:
If your goal here is to bypass the DB config screen and test it immediately, I'd rather actually test the connection here instead of adding GET support to the DB test route. You could test the connection here with the `DB::RAW('SELECT VERSION();');` from below since Laravel will have already set up the DB config in the case that `DATABASE_URL` is already present, and if it succeeds, redirect to the next step immediately. If it fails, then you can fall through to showing the database config form below.

Reply:
it means having duplicate code / bloat. This already needs a re-factoring run-through; possibly class extraction & tests.